### PR TITLE
Uncategorised chars

### PR DESCRIPTION
**Pull request by RoelN**
This will deal with chars that don't match any of the categories in GlyphData. This was needed to catch chars for scripts like Sora Sompeng (in Noto Sans Sora Sompeng), otherwise Specimen Skeleton would be unaware of these characters.

Best option would be to add Sora Sompeng to GlyphData, which will happen after finding a proper "single source of truth" place for this data. (Original XML file?) But a fallback catching all unrecognised chars is good to have anyway.

### CODE REVIEW

File: fontData.js [hunk 1]
```diff
@@ -192,6 +199,21 @@ const buildChars = font => {
     }
   }
 
+  // List all chars not grouped under scripts in a misc category
+  // Also, ignore 0xFFFF which is erroneously reported as a char
+  // by Fontkit
```
Comment (pascalw) [inline fontData.js:204]:
```suggestion
  // by Font.js
```

File: fontData.js [hunk 2]
```diff
@@ -192,6 +199,21 @@ const buildChars = font => {
...
+  const uncategorisedChars = fontCharset.filter(
```
Comment (pascalw) [inline fontData.js:205]: Can you add a test for this please?

Comment (pascalw) [reply fontData.js:205]: There's probably an opportunity here to do this all in one pass; loop over all chars in the font and assign the correct category or mark uncategorised.
For now I guess this is fine as this processing is not done very often. In WF it is probably worth it to refactor this to reduce to a single pass.

File: fontData.js (hunk 1 above)
Comment (RoelN) [reply fontData.js:204]: That's true for the _other_ project ;-) We're still on Fontkit here.

(But I'd more and more love to use the Wakamai Fondue engine here!!)

Comment (pascalw) [reply fontData.js:204]: Oh haha yes you're right, I'm mixing things up here 😅

File: fontData.js (hunk 2 above)
Comment (RoelN) [reply fontData.js:205]: Added tests. Needed a new test font for this.

I haven't ported this to WF yet. (Wouldn't have to if we put this in the engine! ✊ ) But agree that it shouldn't be this crude then.

**Review by pascalw — APPROVED** [review]
👍

**Pull request merged**